### DISCUSSION TITLE
How to bifurcate the UPLOAD_LOCATION enviornment variable into seperate directories for original content and generated content?

### DISCUSSION THREAD

**Discussion by charles-997** [immich-app/immich#12249] [Q&A]
I am setting up Immich and am planning my backup strategy.  [This page within the docs](https://immich.app/docs/administration/backup-and-restore/#filesystem) shows that the `UPLOAD_LOCATION` includes both original content and generated content.

The generated content does not need to be backed up - if it were lost, Immich would rebuild it.

I would like to have two directories on my host system 1) `immich` and 2) `immich_no-backup`

Does anyone have advice on how I can setup my docker-compose to enable the above?  This will allow me to very easily backup the data that needs to be to my offsite NAS and exclude the unnecessary data.

### Proposed Method

My initial thinking is that I can use bind mounts as follows
```
- immich:/usr/src/app/upload/library
- immich:/usr/src/app/upload/upload
- immich:/usr/src/app/upload/profile
```

These are the three sub-directories of `${UPLOAD_LOCATION}` that according to the docs contain original content.

Does anyone know the names of the sub-directories that contain generated content?  I can bind those to `immich_no-backup`.

Please let me know if my proposed method is not best practice, my goal here is simply to separate data on the host that requires backup from data that does not.

**Comment by radh21301** [comment]
Your bind mounts wont work...you cant assign three folders in the docker container to the same folder on the host - you might be able to, but it work very well. 

Best separate them out:
```
- <your base directory>/immich/library:/usr/src/app/upload/library
- <your base directory>/immich/upload:/usr/src/app/upload/upload
- <your base directory>/immich/profile:/usr/src/app/upload/profile

# The folders that immich generates are 'encoded-video' and 'thumbs'
- <your base directory>/immich_no-backup/encoded-video:/usr/src/app/upload/encoded-video
- <your base directory>/immich_no-backup/thumbs:/usr/src/app/upload/thumbs
```

**Reply by charles-997** [reply]
Right - yes. Thank you for catching that typo.  Thank you also for pointing out the folder names for generated content.

**Comment by bo0tzz** [comment]
https://immich.app/docs/guides/custom-locations
